src/components/WalletSelector/multichain/tezos/TezosAddWallet.tsx

```diff
@@ -0,0 +1,261 @@
+import { Button } from 'components/core/Button/Button';
+import colors from 'components/core/colors';
+import { BaseM, TitleS } from 'components/core/Text/Text';
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import styled from 'styled-components';
+import { Web3Error } from 'types/Error';
+import Spacer from 'components/core/Spacer/Spacer';
+import {
+  ADDRESS_ALREADY_CONNECTED,
+  CONFIRM_ADDRESS,
+  INITIAL,
+  PendingState,
+  PROMPT_SIGNATURE,
+} from 'types/Wallet';
+import { useModalActions } from 'contexts/modal/ModalContext';
+import {
+  isEarlyAccessError,
+  useTrackAddWalletAttempt,
+  useTrackAddWalletError,
+  useTrackAddWalletSuccess,
+} from 'contexts/analytics/authUtil';
+import { captureException } from '@sentry/nextjs';
+import { TezosAddWalletFragment$key } from '__generated__/TezosAddWalletFragment.graphql';
+import { removeNullValues } from 'utils/removeNullValues';
+import { useFragment } from 'react-relay';
+import { graphql } from 'relay-runtime';
+import useAddWallet from 'components/WalletSelector/mutations/useAddWallet';
+import useCreateNonce from 'components/WalletSelector/mutations/useCreateNonce';
+import { normalizeError } from '../normalizeError';
+import { WalletError } from '../WalletError';
+import { DAppClient } from '@airgap/beacon-sdk';
+import { generatePayload, getNonceNumber } from './tezosUtils';
+
+type Props = {
+  queryRef: TezosAddWalletFragment$key;
+  reset: () => void;
+};
+
+let beaconClient: DAppClient;
+
+if (typeof window !== 'undefined') {
+  beaconClient = new DAppClient({ name: 'Gallery' });
+}
```

Comment:
i wonder if it matters if this client is different from the one that's created for the regular login flow?

e.g. user logs in with tezos => first DAppClient created => user adds tezos wallet => 2nd DAppClient created

i don't think this would break anything, but ideally these are shared via React Context